operations/helm/charts/mimir-distributed/values.yaml

```diff
@@ -1581,3 +1581,79 @@ gateway:
       - secretName: gem-gateway-tls
         hosts:
           - gateway.gem.example.com
+
+metaMonitoring:
+  grafanaAgent:
+    # -- Controls whether to create PodLogs, MetricsInstance, LogsInstance, and GrafanaAgent CRs to scrape the
+    # ServiceMonitors of the chart and ship metrics and logs to the remote endpoints below.
+    # Note that you need to configure serviceMonitor in order to have some metrics available.
+    enabled: false
+
+    # -- Controls whether to install the Grafana Agent Operator and its CRDs.
+    # Note that helm will not install CRDs if this flag is enabled during an upgrade.
+    # In that case install the CRDs manually from https://github.com/grafana/agent/tree/main/production/operator/crds
+    installOperator: false
+
+    # -- Controls whether to scrape cadvisor, kubelet, and kube-state-metrics for metrics related to Mimir.
+    # The scraped metrics will be limited to ones concerning Mimir only: where pod
+    scrapeK8s: true
+
+    logs:
+      # -- Default destination for logs. The config here is translated to Promtail client
+      # configuration to write logs to this Loki-compatible remote. Optional.
+      remote:
+        url: ''
+
+        # -- Used to set X-Scope-OrgID header on requests. Usually not used in combination with basic authentication.
+        tenantId: ''
+        auth:
+          # -- Basic authentication username. Optional.
+          username: ''
+
+          # -- The value under key 'password' in this secret will be used as the basic authentication password. Optional.
+          passwordSecretName: ''
+
+      # -- Client configurations for the LogsInstance that will scrape Mimir pods. Follows the format of .remote.
+      additionalClientConfigs: [ ]
+
+    metrics:
+      # -- Default destination for metrics. The config here is translated to remote_write
+      # configuration to push metrics to this Prometheus-compatible remote. Optional.
+      # Note that you need to configure serviceMonitor in order to have some metrics available.
+      remote:
+        url: ''
+
+        # -- Used to add HTTP headers to remote-write requests.
+        headers: { }
+        auth:
+          # -- Basic authentication username. Optional.
+          username: ''
+
+          # -- The value under key 'password' in this secret will be used as the basic authentication password. Optional.
+          passwordSecretName: ''
+
+      # -- Additional remote-write for the MetricsInstance that will scrape Mimir pods. Follows the format of .remote.
+      additionalRemoteWriteConfigs: [ ]
```

Comment:
If I try to add a new config like this:

```
      additionalRemoteWriteConfigs:
      - url: 'http://gem-mimir-gateway.default.svc.cluster.local/api/prom/push'
        headers:
          X-Scope-OrgID: oss-self-monitoring
```

I get an error message:

```
Error: UPGRADE FAILED: template: mimir-distributed/templates/metamonitoring/metrics-instance.yaml:24:94: executing "mimir-distributed/templates/metamonitoring/metrics-instance.yaml" at <.auth.username>: nil pointer evaluating interface {}.username
```

I can't seem to find any set of values that works here. 

```
      additionalRemoteWriteConfigs:
      - url: 'http://gem-mimir-gateway.default.svc.cluster.local/api/prom/push'
        headers:
          X-Scope-OrgID: oss-self-monitoring
        auth:
          username: "something"
          passwordSecretName: "agent-password"
```

gives me this:

```
Error: UPGRADE FAILED: error validating "": error validating data: ValidationError(MetricsInstance.spec.remoteWrite[1]): unknown field "externalLabels" in com.grafana.monitoring.v1alpha1.MetricsInstance.spec.remoteWrite
```

Reply:
aii, that looks bad. should be fixed by two commits I pushed